lua_cjson.c

```diff
@@ -1077,13 +1093,18 @@ static int json_is_invalid_number(json_parse_t *json)
 static void json_next_number_token(json_parse_t *json, json_token_t *token)
 {
     char *endptr;
-
-    token->type = T_NUMBER;
-    token->value.number = fpconv_strtod(json->ptr, &endptr);
-    if (json->ptr == endptr)
+    token->value.integer = strtoll(json->ptr, &endptr, 0);
```

Comment:
This call is (mostly) wasted for floating point number. I'm not sure if it's worth it for Lua 5.1.

Reply:
I'm using it in skynet(Lua5.3.4), It's sure that is worth. But in Lua5.1, it doesn't break the compatibility.

Reply:
@korialuo What I'm saying is that it unnecessary slows down the parsing of every JSON floating point number for Lua 5.1 (and LuaJIT), which is not desired. We can definitely improve things here specifically for Lua 5.1 while still supporting later Lua language versions.